src/main/java/org/terasology/launcher/repositories/JenkinsClient.java

```diff
@@ -32,22 +33,43 @@
         this.gson = gson;
     }
 
+    /**
+     * Open an input stream from the given URL with a default timeout configured.
+     *
+     * The shorthand {@link URL#openStream()} does not allow for setting a connection timeout. Therefore, we interleave
+     * the original implementation {@code openConnection().getInputStream()} with a step to configure a default timeout.
+     *
+     * The default timeout is 10 seconds.
+     *
+     * @param url the URL to open the stream for
+     * @return an input stream similar to {@code url.openStream()} with a connection timeout of 10 seconds
+     * @throws IOException if opening the URL connection or retrieving the input stream fails
+     */
+    static InputStream openStream(URL url) throws IOException {
+        // this is a static member to indicate that it is independent of the client itself, and to cleanly stub it for
+        // testing purposes without the need to mock the class-under-test itself.
+        URLConnection connection = url.openConnection();
+        connection.setConnectTimeout(10 * 1000);
+        return connection.getInputStream();
+    }
+
     Jenkins.ApiResult request(URL url) {
         Preconditions.checkNotNull(url);
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(openStream(url)))) {
             return gson.fromJson(reader, Jenkins.ApiResult.class);
         } catch (JsonSyntaxException | JsonIOException e) {
             logger.warn("Failed to read JSON from '{}'", url.toExternalForm(), e);
         } catch (IOException e) {
-            logger.warn("Failed to read from URL: {}", url.toExternalForm(), e);
+            logger.warn("Failed to read from URL '{}'\n\t{}", e.getMessage(), url.toExternalForm());
```

Comment:
I think you've gotten the argument order swapped here. You still want the url first,  
and then if you want to just do e.getMessage() instead of the logger's default exception formatter, that's your second argument.

Reply:
(also, if you find yourself avoiding the logger's default exception formatter often, that could be a sign that we should reconfigure the logger)

Reply:
Hm, I'm actually not sure about best logging patterns. If you happen to know good resources on that let me know :nerd_face: 

The arguments are actually in the order I want them to be. The message is usually short and descriptive, so you can see what's wrong directly. The URL is pretty lengthy (maybe another option would be to drop the query parameters?), that's I wanted to move that to second line. Just forgot to adjust the text itself to reflect that.

On changing the default exception formatter: I do think that logging a full stack trace is only appropriate if we expect this is needed for investigating the issue, and/or if this is fatal error we cannot recover from. In this case, an exception here is unfortunate, but can happen for various reasons. Usually the stacktrace does not give helpful info here, but clutters the log.

Reply:
ah, I didn't expect the URL to be that long, but that would explain it

Reply:
For reference the full URL with search query parameters:

```
http://jenkins.terasology.io/teraorg/job/Terasology/job/Omega/job/master/api/json?tree=builds[number,timestamp,result,artifacts[fileName,relativePath],url]
```

Consisting of the base URL

```
http://jenkins.terasology.io/
```

followed by the job descriptor

```
/teraorg/job/Terasology/job/Omega/job/master
```

and finally a _filtered view_ on the data

```
/api/json?tree=builds[number,timestamp,result,artifacts[fileName,relativePath],url]
```

So yeah, it's pretty lengthy :sweat_smile: